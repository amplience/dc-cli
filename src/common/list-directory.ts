import * as fs from 'fs';

export const listDirectory = (dir: string): string[] => {
  try {
    return fs.readdirSync(dir);
  } catch (err) {
    throw new Error(`Unable to read ${dir}`);
  }
};
