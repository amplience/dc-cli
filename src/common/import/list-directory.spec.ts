import fs from 'fs';
import { listDirectory } from './list-directory';

jest.mock('fs');

describe('list-directory', () => {
  describe('listDirectory', () => {
    afterEach((): void => {
      jest.resetAllMocks();
    });
    it('should return a list of files from a directory', () => {
      const directoryFileNames = ['file-a.json', 'file-b.json'];
      (fs.readdirSync as jest.Mock).mockReturnValue(directoryFileNames);
      const result = listDirectory('./dir-name');
      expect(result).toEqual(directoryFileNames);
    });

    it('should throw an error when directory read fails', () => {
      (fs.readdirSync as jest.Mock).mockImplementationOnce(() => {
        throw new Error('Directory does not exist');
      });
      expect(() => listDirectory('./dir-name')).toThrowErrorMatchingSnapshot();
    });
  });
});
