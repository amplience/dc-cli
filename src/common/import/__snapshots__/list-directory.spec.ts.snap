// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-directory listDirectory should throw an error when directory read fails 1`] = `"Unable to read ./dir-name"`;
